test/models/driver_test.rb

```diff
@@ -54,28 +54,65 @@
 
   # Tests for methods you create should go here
   describe "custom methods" do
     describe "average rating" do
       # Your code here
+      it "accurately calculates the average rating of multiple trips" do
+        new_driver.save
+        new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 5, cost: 1234)
+        trip_2 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 3, cost: 6334)
+        trip_3 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 1, cost: 1234)
+        trip_4 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 4, cost: 6334)
+
+        expect(new_driver.average_rating).must_be_close_to 3.3
+      end
+
+      it "accurately calculates the average rating of multiple trips even if there are nil ratings" do
+        new_driver.save
+        new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 5, cost: 1234)
+        trip_2 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 3, cost: 6334)
+        trip_3 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: nil, cost: 1234)
+        trip_4 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 4, cost: 6334)
+
+        expect(new_driver.average_rating).must_be_close_to 4
+      end
     end
 
     describe "total earnings" do
-      # Your code here
+      it "accurately calculates the total earnings of multiple trips" do
+        new_driver.save
+        new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 5, cost: 12.34)
+        trip_2 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 3, cost: 20.50)
+        trip_3 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 1, cost: 15.65)
+        trip_4 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 4, cost: 63.17)
+
+        expect(new_driver.total_earnings).must_be_close_to 84.05
+      end
     end
 
     describe "can go online" do
-      # Your code here
+      it "can change availability to true from false" do
+        new_driver.available = false
+        new_driver.set_available
+        expect(new_driver.available).must_equal true
+      end
     end
 
     describe "can go offline" do
-      # Your code here
-    end
+      it "can change availability to false from true" do
+        new_driver.save
```

Comment:
Veeerryyyy small note: I think I might understand why you all included this line-- to make it clear that `available` is `true`. However, from my point of view, if you all had something like the `set_available` test, I'd think that was totally reasonable.

```suggestion
        new_driver.available = true
```

An even weirder thing you could do is make your "Arrange" step an assert that checks that `available` isn't false, in case you wanted to assume a default value of `true`

```suggestion
        expect(new_driver.available).must_equal true
```